tests/late_joiners.py

```diff
@@ -104,107 +107,106 @@ def run_requests(nodes, total_requests, start_id, final_msg, final_msg_id):
                 node_id += 1
                 c = clients[node_id % len(clients)]
                 try:
-                    c.rpc("LOG_record", {"id": id, "msg": long_msg})
+                    check_commit(
+                        c.rpc("LOG_record", {"id": id, "msg": long_msg}), result=True
+                    )
                 except (TimeoutError, requests.exceptions.ReadTimeout,) as e:
```

Comment:
It seems that `requests.exceptions.ReadTimeout` is caught by `clients.py` and raises a `TimeoutError`. I think it would be better to remove the `cant_fail` argument altogether and catch the `TimeoutError` in the caller.

Reply:
In the caller of the `run_requests` method? The `cant_fail` argument is there to make sure that failures are being risen on normal request execution, but signifies that timeout errors can (and will occur) when the nodes are suspended, and that that is OK.

Reply:
I see - the loop will continue if this fails 👍 I think the `requests.exceptions.ReadTimeout` can be removed still though.

Reply:
Yup we want to keep spamming the nodes until they respond again. I'll check to see if it can be removed, but I think it still gets thrown.

Reply:
`requests.exceptions.ReadTimeout` still bubbles up here occasionally so we will keep it.